resources/lib/ttml2srt.py

```diff
@@ -27,205 +27,224 @@
-        infile(str): full name of the input file
-
-        outfile(str): full name of the converted file
-    """
-    tree = ET.parse(infile)
-    root = tree.getroot()
-
-    # strip namespaces
-    for elem in root.getiterator():
-        elem.tag = elem.tag.split('}', 1)[-1]
-        elem.attrib = {name.split('}', 1)
-                       [-1]: value for name, value in list(elem.attrib.items())}
-
-    # get styles
-    styles = {}
-    for elem in root.findall('./head/styling/style'):
-        style = {}
-        if 'color' in elem.attrib:
-            color = elem.attrib['color']
-            if color not in ('#FFFFFF', '#000000'):
-                style['color'] = color
-        if 'fontStyle' in elem.attrib:
-            fontstyle = elem.attrib['fontStyle']
-            if fontstyle in ('italic', ):
-                style['fontstyle'] = fontstyle
-        styles[elem.attrib['id']] = style
-
-    body = root.find('./body')
-
-    # parse correct start and end times
-    def _parse_time_expression(expression, default_offset=timedelta(0)):
-        offset_time = re.match(
-            r'^([0-9]+(\.[0-9]+)?)(h|m|s|ms|f|t)$', expression)
-        if offset_time:
-            time_value, _, metric = offset_time.groups()
-            time_value = float(time_value)
-            if metric == 'h':
-                return default_offset + timedelta(hours=time_value)
-            elif metric == 'm':
-                return default_offset + timedelta(minutes=time_value)
-            elif metric == 's':
-                return default_offset + timedelta(seconds=time_value)
-            elif metric == 'ms':
-                return default_offset + timedelta(milliseconds=time_value)
-            elif metric == 'f':
-                raise NotImplementedError(
-                    'Parsing time expressions by frame is not supported!')
-            elif metric == 't':
-                raise NotImplementedError(
-                    'Parsing time expressions by ticks is not supported!')
-
-        clock_time = re.match(
-            r'^([0-9]{2,}):([0-9]{2,}):([0-9]{2,}(\.[0-9]+)?)$', expression)
-        if clock_time:
-            hours, minutes, seconds, _ = clock_time.groups()
-            return timedelta(hours=int(hours), minutes=int(minutes), seconds=float(seconds))
-
-        clock_time_frames = re.match(
-            r'^([0-9]{2,}):([0-9]{2,}):([0-9]{2,}):([0-9]{2,}(\.[0-9]+)?)$', expression)
-        if clock_time_frames:
-            raise NotImplementedError(
-                'Parsing time expressions by frame is not supported!')
-
-        raise ValueError('unknown time expression: %s' % expression)
-
-    def _parse_times(elem, default_begin=timedelta(0)):
-        if 'begin' in elem.attrib:
-            begin = _parse_time_expression(
-                elem.attrib['begin'], default_offset=default_begin)
-        else:
-            begin = default_begin
-        elem.attrib['{abs}begin'] = begin
+from xml.etree import ElementTree as ET
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Using ElementTree to parse untrusted XML data is known to be vulnerable to XML attacks. Replace ElementTree with the equivalent defusedxml package, or make sure defusedxml.defuse_stdlib() is called.](https://app.codacy.com/manual/MediathekView/plugin.video.mediathekview/pullRequest?prid=5447573)